### PR TITLE
Separate pt-BR and pt-PT through a crowdin custom mapping

### PR DESCRIPTION
**Pull request by oliverguenther**
Previously, both pt-BR and pt-PT in crowdin were active but mapped to the same pt.yml file. pt-BR "won" that and resulted in the only portuguese language being available.

On Crowdin, I added a custom mapping that would export both in the above locales like we already did for zh-TW and zh-CN.  It requires a tiny migration to replace previous occurrrences of pt in user languages and default locales.

While I was adding to the default set of languages, I went through the list in crowdin and added all that are somewhat translated. Although I do believe we should add all languages (possibly with a warning if they are not translated well so far).

https://community.openproject.org/work_packages/53374/activity

To be able to reason about the default language selection, I've added that as a "Feature" so it turns up in the release notes, even though it's just that one line change.
https://community.openproject.org/work_packages/53378

### CODE REVIEW

**Review by ulferts — APPROVED** [review]

**Comment by cbliard** [comment]
@ulferts please do not merge yet, I'm reviewing too.

File: config/constants/settings/definition.rb [hunk 1]
```diff
@@ -147,7 +147,9 @@ class Definition
       },
       available_languages: {
         format: :array,
-        default: %w[en de fr es pt it zh-CN ko ru].freeze,
+        # Manually managed list with languages that have ~50+ translation ratio in Crowdin
+        # https://crowdin.com/project/openproject
+        default: %w[ca cs de el en es fr hu id it ja ko ko lt nl no pl pt-BR pt-PT ro ru sk sl sv tr uk zh-CN zh-TW].freeze,
```
Comment (cbliard) [inline config/constants/settings/definition.rb]:
```suggestion
        default: %w[ca cs de el en es fr hu id it ja ko lt nl no pl pt-BR pt-PT ro ru sk sl sv tr uk zh-CN zh-TW].freeze,
```

Also, sk is only 36% but is included. Is it intended?

File: config/locales/generated/pt-BR.yml [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 # The translations come from version 42 of the Unicode CLDR project .
 #
 # The Unicode Common Locale Data Repository (CLDR) provides key building
 # blocks for software to support the world's languages, with the largest
 # and most extensive standard repository of locale data available.
 ---
-pt:
+pt-BR:
   cldr:
     language_name: Português
```
Comment (cbliard) [inline config/locales/generated/pt-BR.yml]: It feels wrong. It should mention Brazil somewhere.

Actually, in the CLDR, there is no language information for pt-BR from file `common/main/pt_BR.xml`, and the way the script works fall back to `pt` locale from `common/main/pt.xml` file instead. That's why it's "Português".

The translations for variations "pt-BR" and "pt-PT" do not exist in `common/main/pt.xml`, but do exist in `common/main/pt_PT.xml`:

```
  <language type="pt_BR">português do Brasil</language>
  <language type="pt_PT">português europeu</language>
```

The correct language name should be "Português do Brasil".

We could also use the "territory" information from `common/main/pt.xml` to generate the names:
```
  <territory type="BR">Brasil</territory>
  ...
  <territory type="PT">Portugal</territory>
```

Names could then be "Português (Brasil)" and "Português (Portugal)".

Which one would be better?

The script should be modified to be better at automatically resolving those edge cases.

**Review by cbliard — CHANGES_REQUESTED** [review]
With the proposed changes, the language name for "pt-BR" would be "Português".
It should be "Português do Brasil" ot "Português (Brasil)".

I'll have a look at the generation script to see if it can resolve this properly.

File: config/locales/generated/pt-BR.yml (hunk 2 above)
Comment (cbliard) [reply config/locales/generated/pt-BR.yml]: I modified the script to use "Português do Brasil" instead and commited it.

**Comment by cbliard** [comment]
@ulferts you can review a second time now, and merge if ok.

**Comment by oliverguenther** [comment]
Thanks @cbliard, I verified your changes

**Pull request merged**